pkg/controller/operators/catalog/operator.go

```diff
@@ -727,10 +727,19 @@ func (o *Operator) syncRegistryServer(logger *logrus.Entry, in *v1alpha1.Catalog
 
 	// requeue the catalog sync based on the polling interval, for accurate syncs of catalogs with polling enabled
 	if out.Spec.UpdateStrategy != nil {
-		logger.Debugf("requeuing registry server sync based on polling interval %s", out.Spec.UpdateStrategy.Interval.Duration.String())
-		resyncPeriod := reconciler.SyncRegistryUpdateInterval(out, time.Now())
-		o.catsrcQueueSet.RequeueAfter(out.GetNamespace(), out.GetName(), queueinformer.ResyncWithJitter(resyncPeriod, 0.1)())
-		return
+		if out.Spec.UpdateStrategy.RegistryPoll != nil {
+			if len(out.Spec.UpdateStrategy.RegistryPoll.ParsingError) != 0 {
```

Comment:
since `ParsingError` is a string type is there a benefit to checking `len(string) != 0` versus checking the string directly via `string != ""`?

Reply:
That's a good point @exdx . I can't remember why I'd checked for the length instead of checking for the zero value, possibly an oversight. I've updated the code to check for the zero value.